CIP-XXXX/README.md

```diff
@@ -0,0 +1,190 @@
+
+#### cardano.{walletName}.enable(extensions: Extension[] = [{ cip: 30 }]): Promise\<API>
+
+Errors: `ConnectorError`
+
+This is the entrypoint to start communication with the user's wallet. The wallet should request the user's permission to connect the web page to the wallet, and if permission has been granted, the [Extendable API](#extendable-api) will be returned to the dApp to use. The wallet can choose to maintain a whitelist to not necessarily ask the user's permission every time access is requested. But this behavior is up to the wallet and should be transparent to web pages using this API.
+
+Upon start, dApps can explicitly request a list of additional functionalities they expect as a list of CIP numbers capturing those extensions. Note that it's possible for two extensions to be mutually incompatible (because they provide two conflicting features). While we may try to avoid this as much as possible while designing CIPs, it is also the responsibility of wallet providers to assess whether they can support a given combination of extensions, or not. Hence, wallets aren't expected to fail should they not recognize or not support a particular combination of extensions. As a result, dApps may fail and inform their users or may use a different, less-efficient, strategy to cope with a lack of functionality.
+
+Once a connection is established, this can be called multiple times by the dApp to request access to further extensions. User permission must be asked if new extensions are requested. If this is called without new extensions requests then no user permission is required.
+
+**Note**: In order to improve backwards compatibility an empty `extensions` field should supply CIP-30 functionality, as if a CIP-30 `extension` object was provided.
+
+#### cardano.{walletName}.isEnabled(): Promise\<Extension[]>
+
+Errors: `ConnectorError`
+
+Returns a list of extensions which have been enabled by the wallet. If no wallet access has been granted then a `ConnectorError` with  `Refused` error code should be thrown.
+
+#### cardano.{walletName}.supportedExtensions: Extension[] | undefined
+
+A list of extensions supported by the wallet. Extensions may be requested by dApps on initialization. Dapps can repeatably call `wallet.enable()` to add further extensions. Some extensions may be mutually conflicting, and this list does not thereby reflect what extensions will be enabled by the wallet. Yet it informs on what extensions are known and can be requested by dApps if needed.
+
+`undefined` is equivalent to `{cip: 30}`.

```

Comment:
I think this is redundant.

I do not see much utility in fine grain switching of extensions.  
If the process was just:

```
cardano.{walletName}.enable();
extensions = cardano.{walletName}.isEnabled();
```

That tells you everything you need as a dApp.
1. Does this wallet support this cip, or is it just obeying cip-30.
2. What extensions does the wallet support.
3. All supported extensions are enabled.

Reply:
I do agree with this, I think I will do a revision of this proposal.